src/Grid/index.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+import styled, { css } from 'styled-components';
+import { LayoutProps, Style } from '../types';
+import { createResponsiveStyles, baseProperty, layout, responsiveProperty } from '../utils';
+
+export type GridProps = {
+   className?: string;
+   templateRows?: Style<React.CSSProperties['gridTemplateRows']>;
+   templateColumns?: Style<React.CSSProperties['gridTemplateColumns']>;
+   gap?: Style<string | number>;
+} & LayoutProps;
+
+const gridStyles = css`
+   ${baseProperty('templateRows', 'grid-template-rows')};
+   ${baseProperty('templateColumns', 'grid-template-columns')};
+   ${baseProperty('gap', 'row-gap')};
+   ${baseProperty('gap', 'column-gap')};
+   ${createResponsiveStyles(
+      (key) => css`
+         ${responsiveProperty(key, 'templateRows', 'grid-template-rows')};
+         ${responsiveProperty(key, 'templateColumns', 'grid-template-columns')};
+         ${responsiveProperty(key, 'gap', 'row-gap')};
+         ${responsiveProperty(key, 'gap', 'column-gap')};
+      `
+   )}
+`;
+
+export const Grid = styled.div.withConfig({
+   shouldForwardProp: (prop, defaultValidator) => ![''].includes(prop) && defaultValidator(prop),
```

Comment:
It was meant to be filled with the props that should not be passed to the dom element. Again, I haven't polished all this stuff because I was eager to get a first version working and then refine it